### ISSUE TITLE
[macOS] No matching version for dev.nokee.framework:JavaVM:[10.14,)

### ISSUE THREAD

**Issue by weisJ** [nokeedev/gradle-native#44]
Using the latest nightly build `0.4.0-00ed537` trying to build a macOS library using
````kotlin
nativeImplementation("dev.nokee.framework:JavaVM:[10.14,)")
````
the build fails, claiming that no matching version could be found.

Some context:
On the runner there are multiple two SDK versions installed `10.14.*` and `10.15.*`.
The `10.14.*` variant can be used by switching to XCode `10.3`. For the system then only the `10.14.*` SDKs are visible. But the plugin can't seem to find the SDK version.

Maybe it is still looking in the `10.15` install location?

Edit: Yes this seems to be the case:
````
Starting process 'command '/usr/bin/clang++''. Working directory: /usr/bin Command: /usr/bin/clang++ -m64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -dM -E -v -
````

I am doing this because using the newer SDK version, even if `-mmacosx-version-min=10.14` is specified the resulting library lists `10.15` as the minimum supported OS version.

````
       cmd LC_BUILD_VERSION
   cmdsize 32
  platform 1
       sdk 10.15.4
     minos 10.15    <----
    ntools 1
      tool 3
   version 556.6
```` 

[Build Log](https://github.com/weisJ/auto-dark-mode/runs/704298785?check_suite_focus=true)

**Comment by lacasseio** [comment]
Sorry for missing this issue. I'm assuming it's related to the mismatch version between buildSrc and the main build. I saw a similar problem when I worked on https://github.com/weisJ/auto-dark-mode/pull/5. What I was seeing was a nightly version 0.3 of `platformJni` used in buildSrc which was also used in the main build. I pushed a change to release an all-inclusive artifact to avoid this kind of mismatch: https://github.com/nokeedev/gradle-native/issues/46. I can incorporate it tomorrow in the PR. In the meantime, can you validate the PR works as intended?

**Comment by lacasseio** [comment]
I pushed a commit to the PR to use the all-inclusive artifact in buildSrc. I'm looking at the other failures you are seeing.

**Comment by lacasseio** [comment]
To circle back on this issue with more context, Nokee seems to be misbehaving with either Xcode 10.3 or how Xcode 10.3 is selected. I haven't put much work yet into auto discovering Xcode installation, yet. However, with what I have seen in the wild, the pattern for versioned Xcode like the following seems to be quite frequent: `/Application/Xcode_<version>.app` We wouldn't trust the version in the file name but we could detect the presence of them and use `DEVELOPER_DIR` with `xcrun` to query the information like SDK version/path and frameworks. It would avoid having to jump through the hoops of configuring the environment.

In your case, it seems you really want to use version 10.14 which seems to only be available in 10.3. I would probably configure some dependency rules to relax the version constraint on developer machines and ensure the minimum OS version is selected for the CI build. It would be aligned with the current "use prebuilt binaries when unbuildable" use case to ease up the development.

**Comment by lacasseio** [comment]
The issue was around the lack of modelling for the SDK. By default `xcrun` will assume `-sdk` for the current OS. Nokee now parses the output for `xcodebuild -showsdks`. It limits the framework search to macOS SDK. The other SDK will be handled during iOS improvements.
